tests/pkio_test.py

```diff
@@ -116,7 +116,7 @@ def test_walk_tree_and_sorted_glob():
             'When walking tree with file_re, should only return matching files'
         assert [expect[0]] == list(pkio.walk_tree('.', '^d1')), \
             'When walking tree with file_re, file to match does not include dir being searched'
-        assert pkio.sorted_glob('*[42]') == [py.path.local(f) for f in ('d2', 'd4')]
+        assert pkio.sorted_glob('*[42]', key='mtime') == [py.path.local(f) for f in ('d4', 'd2')]
```

Comment:
mtime is not going to be very good for this situation, because all files are created at the same time and Linux only keeps mtime in seconds. This would be better:

```py
pkio.sorted_glob('*/*/*', key='basename') == expect
```

Reply:
basename is a problem but it's a good test, because it is a useful. It is not a callable so the key code just needs to test whether the object is callable or not.

Reply:
Good thinking. 66fb4e1